### PR TITLE
Implement formatting of impls

### PR DESCRIPTION
**Pull request by nrc**

### CODE REVIEW

File: src/items.rs [hunk 1]
```diff
@@ -435,6 +435,104 @@ impl<'a> FmtVisitor<'a> {
...
+
+            result.push_str(trim_newlines(&visitor.buffer.to_string()));
+            result.push('\n');
+            result.push_str(&indent_str);
+        }
+        result.push('}');
+
+        Some(result)
+    } else {
+        unreachable!();
```
Comment (marcusklaas) [inline src/items.rs]: Why can't we just push a `}` to result and be done here?

Comment (marcusklaas) [inline src/items.rs]: I mean, this is reachable for marker traits, right?

Comment (nrc) [inline src/items.rs]: This is the scope for the `if let` at the top, not for the items.is_empty thing

**Comment by nrc** [comment]
Rebased, just a bug fix commit now

**Pull request merged**